unit-tests/syncer/test-ts-diff-fps.py

```diff
@@ -0,0 +1,258 @@
+
+# D  C  @timestamp
+# -- -- -----------
+# 0     @0
+# 1     @10
+#    0  @0
+#
+generate_depth_frame( frame_number = 0, timestamp = gap_d * 0 )
+generate_depth_frame( 1, gap_d * 1 )  # @10
+generate_color_frame( 0, gap_c * 0 )  # @0 -- small latency
+expect( depth_frame = 0 )                          # syncer doesn't know about color yet
+expect( depth_frame = 1 )
+#expect( color_frame = 0, nothing_else = True )
+# We'd expect C0 to not wait for another frame, but it does: @0 is comparable to @20 (D.NE)
+# because it's using C.fps (10 fps -> 100 gap / 2 = 50ms error, so 0~=20)!
+expect_nothing()
+
+# 2     @20
+generate_depth_frame( 2, gap_d * 2 )
+expect( depth_frame = 2, color_frame = 0, nothing_else = True )
+
+test.finish()
+#
+#############################################################################################
+#
+test.start( "Depth waits for next Color" )
+
+# 3     @ 30
+# 4     @ 40
+# 5     @ 50
+# 6     @ 60
+# 7     @ 70
+# 8     @ 80
+# 9     @ 90
+# 10    @100 -> wait
+# 11    @110
+#
+generate_depth_frame(  3, gap_d *  3 ); expect( depth_frame =  3 )
+generate_depth_frame(  4, gap_d *  4 ); expect( depth_frame =  4 )
+generate_depth_frame(  5, gap_d *  5 ); expect( depth_frame =  5 )
+generate_depth_frame(  6, gap_d *  6 ); expect( depth_frame =  6 )
+generate_depth_frame(  7, gap_d *  7 ); expect( depth_frame =  7 )
+generate_depth_frame(  8, gap_d *  8 ); expect( depth_frame =  8 )
+generate_depth_frame(  9, gap_d *  9 ); expect( depth_frame =  9 )
+generate_depth_frame( 10, gap_d * 10 ); #expect( depth_frame = 10 )
+# C.NE is @100, so it should wait...
+expect_nothing()
+generate_depth_frame( 11, gap_d * 11 );
+expect_nothing()
+
+#    1  @100 -> release (D10,C1) and (D11)
+generate_color_frame(  1, gap_c *  1 );  # @100 -- small latency
+expect( depth_frame = 10, color_frame = 1 )
+expect( depth_frame = 11 )
+expect_nothing()
+
+# 12    @120 doesn't wait
+generate_depth_frame( 12, gap_d * 12 );
+expect( depth_frame = 12 )
+expect_nothing()
+
+test.finish()
+#
+#############################################################################################
+#
+test.start( "Color is early" )
+
+# 13    @130
+# 14    @140
+# 15    @150
+# 16    @160
+# 17    @170
+#    2  @200 -> wait
+# 18    @180 -> (D18,C2) ?! why not wait for (D20,C2) ?!
+#
+generate_depth_frame( 13, gap_d * 13 ); expect( depth_frame = 13 )
+generate_depth_frame( 14, gap_d * 14 ); expect( depth_frame = 14 )
+generate_depth_frame( 15, gap_d * 15 ); expect( depth_frame = 15 )
+generate_depth_frame( 16, gap_d * 16 ); expect( depth_frame = 16 )
+generate_depth_frame( 17, gap_d * 17 ); expect( depth_frame = 17 )
+
+generate_color_frame( 2, gap_c * 2 ); expect_nothing()
+# We're waiting for D.NE @180, because it's comparable (using min fps of the two)
+
+generate_depth_frame( 18, gap_d * 18 )
+# Now we get both back:
+expect( depth_frame = 18, color_frame = 2, nothing_else = True )
+
+# But wait... why match C@200 to D@180 and not wait for D@200??
```

Comment:
I agree , also its not make sense that depth_frame = 2, color_frame = 0 are coupled

Reply:
We can discuss this separately -- but I want to reflect the way things are today before we make internal changes